qiskit/transpiler/preset_passmanagers/level0.py

```diff
@@ -107,98 +88,49 @@ def _choose_layout_condition(property_set):
     elif routing_method == "none":
-        _swap += [
-            Error(
-                msg="No routing method selected, but circuit is not routed to device. "
-                "CheckMap Error: {check_map_msg}",
-                action="raise",
-            )
-        ]
+        routing_pass = Error(
+            msg="No routing method selected, but circuit is not routed to device. "
+            "CheckMap Error: {check_map_msg}",
+            action="raise",
+        )
     else:
         raise TranspilerError("Invalid routing method %s." % routing_method)
 
-    # 5. Unroll to the basis
-    if translation_method == "unroller":
-        _unroll = [Unroller(basis_gates)]
-    elif translation_method == "translator":
-        from qiskit.circuit.equivalence_library import SessionEquivalenceLibrary as sel
-
-        _unroll = [UnrollCustomDefinitions(sel, basis_gates), BasisTranslator(sel, basis_gates)]
-    elif translation_method == "synthesis":
-        _unroll = [
-            Unroll3qOrMore(),
-            Collect2qBlocks(),
-            ConsolidateBlocks(basis_gates=basis_gates),
-            UnitarySynthesis(basis_gates, approximation_degree=approximation_degree),
-        ]
-    else:
-        raise TranspilerError("Invalid translation method %s." % translation_method)
-
-    # 6. Fix any bad CX directions
-    _direction_check = [CheckGateDirection(coupling_map)]
-
-    def _direction_condition(property_set):
-        return not property_set["is_direction_mapped"]
-
-    _direction = [GateDirection(coupling_map)]
-
-    # 7. Unify all durations (either SI, or convert to dt if known)
-    # Schedule the circuit only when scheduling_method is supplied
-    _scheduling = [TimeUnitConversion(instruction_durations)]
-    if scheduling_method:
-        if scheduling_method in {"alap", "as_late_as_possible"}:
-            _scheduling += [ALAPSchedule(instruction_durations)]
-        elif scheduling_method in {"asap", "as_soon_as_possible"}:
-            _scheduling += [ASAPSchedule(instruction_durations)]
-        else:
-            raise TranspilerError("Invalid scheduling method %s." % scheduling_method)
-
-    # 8. Call measure alignment. Should come after scheduling.
-    _alignments = [
-        ValidatePulseGates(
-            granularity=timing_constraints.granularity, min_length=timing_constraints.min_length
-        ),
-        AlignMeasures(alignment=timing_constraints.acquire_alignment),
-    ]
-
     # Build pass manager
-    pm0 = PassManager()
     if coupling_map or initial_layout:
-        pm0.append(_given_layout)
-        pm0.append(_choose_layout, condition=_choose_layout_condition)
-        pm0.append(_embed)
-        pm0.append(_unroll3q)
-        pm0.append(_swap_check)
-        pm0.append(_swap, condition=_swap_condition)
-    pm0.append(_unroll)
+        layout = PassManager()
+        layout.append(_given_layout)
+        layout.append(_choose_layout, condition=_choose_layout_condition)
```

Comment:
IIRC it's because the condition differs between the optimization levels. See my comment below (which I meant to put here not there, but it kind of works for either). Basically I was trying to keep things in common isolated to just the pieces which are shared completely between each optimization level. That way you can got to a specific optimization level's code and see what is shared and what is unique.